tests/build/build_templates.go

```diff
@@ -179,10 +187,37 @@ var _ = framework.BuildSuiteDescribe("Build templates E2E test", Label("build",
 	defer GinkgoRecover()
 	Describe("HACBS pipelines", Ordered, Label("pipeline"), func() {
 
-		var applicationName, componentName, symlinkComponentName, symlinkPRunName, testNamespace string
+		var applicationName, symlinkPRunName, testNamespace string
+		symlinkComponents := make(map[string]ComponentScenarioSpec)
+		components := make(map[string]ComponentScenarioSpec)
 		var kubeadminClient *framework.ControllerHub
 		var pipelineRunsWithE2eFinalizer []string
 
+		for _, gitUrl := range componentUrls {
+			scenario := GetComponentScenarioDetailsFromGitUrl(gitUrl)
+			Expect(scenario.PipelineBundleNames).ShouldNot(BeEmpty())
+			for _, pipelineBundleName := range scenario.PipelineBundleNames {
+				componentName := fmt.Sprintf("test-comp-%s", util.GenerateRandomString(4))
+
+				s := scenario.DeepCopy()
+				s.PipelineBundleNames = []string{pipelineBundleName}
+
+				components[componentName] = s
+			}
+		}
+
+		symlinkScenario := GetComponentScenarioDetailsFromGitUrl(pythonComponentGitHubURL)
+		Expect(symlinkScenario.PipelineBundleNames).ShouldNot(BeEmpty())
+		for _, pipelineBundleName := range symlinkScenario.PipelineBundleNames {
```

Comment:
Overall looks good to me. Only this symlink scenario is a negative scenario and also uses the `devfile-sample-python-basic` repo `symlink` branch, so it creating two pipelineruns one with `docker-build` and another with `docker-build-oci-ta`, I think we have use a seperate repo for symlink component test. Maybe we can fix that later with separate PR as well

Reply:
@lcarva symlink component needs to be created once with single pipelineBundle `docker-build`, created a fix in [this PR](https://github.com/konflux-ci/e2e-tests/pull/1396)

Reply:
Thanks for taking care of this!